app/services/catalog/submit_order.rb

```diff
@@ -12,7 +12,11 @@ def process
       @order = Order.find_by!(:id => @order_id)
       @order.order_items.each do |order_item|
         raise Catalog::NotAuthorized unless valid_source?(order_item.portfolio_item.service_offering_source_ref)
-        fail_item(order_item) if Catalog::SurveyCompare.any_changed?(order_item.portfolio_item.service_plans)
+
+        if Catalog::SurveyCompare.any_changed?(order_item.portfolio_item.service_plans)
+          order_item.mark_failed("Order Item Failed: Base survey does not match Topology")
+          raise Catalog::InvalidSurvey, "Base survey does not match Topology"
```

Comment:
@lindgrenj6 Should this exception be handled differently in here
https://github.com/RedHatInsights/catalog-api/blob/805502f53838cc6c42115a152c9e5d05c573e5ac/config/initializers/custom_exception_mappings.rb#L24
When a survey matches is that a server side error or a client side error. If we don't capture this error it will return a bad_request error which would mean that it is a user error. But this seems like a refresh error and keeping the templates in sync.

Reply:
I like the idea of that, opened a story here to track it: https://projects.engineering.redhat.com/browse/SSP-1348

I'll update it in a different PR since it is a bit different than what I was taking care of here.